### PR TITLE
feat: onboard olark web device mode

### PR DESCRIPTION
**Pull request by mihir-4116**
## PR Description

- Olark web device mode development

## Notion ticket

Ticket link

## Screenshots

Please add screenshots for any new features or UI bug fixes for the following browsers -

- Chrome
  >
- Firefox
  >
- Safari
  >

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

- - -
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/762)

### CODE REVIEW

File: integrations/Olark/browser.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+      });
+    }
+
+    if (userTraits?.phone) {
+      window.olark("api.visitor.updatePhoneNumber", {
+        phoneNumber: userTraits.phone,
+      });
+    }
+
+    window.olark("api.visitor.updateCustomFields", traits);
```
Comment (Gauravudia) [inline integrations/Olark/browser.js:108]: Here, we are again sending the email/name/phone in traits. Don't we have to exclude this field from traits?

Comment (mihir-4116) [reply integrations/Olark/browser.js:111]: updated

**Review by Gauravudia — APPROVED** [review]

**Review by Gauravudia — CHANGES_REQUESTED** [review]

File: src/integrations/Olark/browser.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  identify(rudderElement) {
+    logger.debug('===In Olark Identify===');
+    const { message } = rudderElement;
+    const { context } = message;
+
+    const userTraits = getDefinedTraits(message);
+
+    if (context.traits) {
+      const { traits } = context;
+      window.olark('api.visitor.updateCustomFields', traits);
```
Comment (shrouti1507) [inline src/integrations/Olark/browser.js:89]: I think the userTraits fields will be repeating here. Shouldn't we remove it?

File: src/integrations/Olark/utils.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  });
+
+  window.olark('api.chat.onMessageToOperator', function () {
+    let eventName = 'Live Chat Message Sent';
+    if (updateEventNames && eventNames?.chatMessageSent) {
+      eventName = eventNames.chatMessageSent;
+    }
+    window.rudderanalytics.track(
+      'Live Chat Message Sent',
+      { eventName },
```
Comment (shrouti1507) [inline src/integrations/Olark/utils.js:31]: same here, event name should be one

File: src/integrations/Olark/utils.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  });
+
+  window.olark('api.chat.onMessageToVisitor', function () {
+    let eventName = 'Live Chat Message Received';
+    if (updateEventNames && eventNames?.chatMessageReceived) {
+      eventName = eventNames.chatMessageReceived;
+    }
+    window.rudderanalytics.track(
+      'Live Chat Message Received',
+      { eventName },
```
Comment (shrouti1507) [inline src/integrations/Olark/utils.js:47]: same here

File: src/integrations/Olark/utils.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+function recordingLiveChatEvents(updateEventNames, standardEventsMap) {
+  const eventNames = standardEventsMap;
+  window.olark('api.chat.onBeginConversation', function () {
+    let eventName = 'Live Chat Conversation Started';
+    if (updateEventNames && eventNames?.startChat) {
+      eventName = eventNames.startChat;
+    }
+    window.rudderanalytics.track(
+      'Live Chat Conversation Started',
+      { eventName },
```
[the remark this answers is not in the document]
Comment (mihir-4116) [reply src/integrations/Olark/utils.js]: updated

File: src/integrations/Olark/utils.js (hunk 3 above)
Comment (mihir-4116) [reply src/integrations/Olark/utils.js]: updated

File: src/integrations/Olark/utils.js (hunk 4 above)
Comment (mihir-4116) [reply src/integrations/Olark/utils.js]: updated

**Review by Gauravudia — APPROVED** [review]

**Review by shrouti1507 — APPROVED** [review]

**Review by shrouti1507 — COMMENTED** [review]
Please wait for anyone from the SDK team to approve

**Review by MoumitaM — APPROVED** [review]

**Pull request merged**